test/view_component/integration_test.rb

```diff
@@ -38,6 +38,31 @@ class IntegrationTest < ActionDispatch::IntegrationTest
     assert_includes inline_response, baseline_response
   end
 
+  test "template changes are reflected on new request on development" do
```

Comment:
👏 nicely done! Would you mind adding a similar integration test proving that template changes are _not_ reflected when `cache_template_loading` is `true`?